service/matching/task_validation.go

```diff
@@ -56,20 +58,26 @@ type (
 	}
 
 	taskValidatorImpl struct {
-		newIOContextFn func() (context.Context, context.CancelFunc)
-		historyClient  historyservice.HistoryServiceClient
+		newIOContextFn  func() (context.Context, context.CancelFunc)
+		clusterMetadata cluster.Metadata
+		namespaceCache  namespace.Registry
```

Comment:
I think we should use "registry" everywhere
```suggestion
		namespaceRegistry  namespace.Registry
```